### PR TITLE
Add: TensorFlow example for semantic segmentation task guide

### PR DESCRIPTION
**Pull request by MKhalusova**
This PR adds a TensorFlow example to the existing [Semantic Segmentation task guide](https://huggingface.co/docs/transformers/main/en/tasks/semantic_segmentation) using the same dataset and fine-tuning steps. 

This example supplements the existing guide and can be helpful to those who choose TensorFlow over PyTorch and would like to use Transformers for semantic segmentation.

### CODE REVIEW

File: docs/source/en/tasks/semantic_segmentation.mdx [hunk 1]
```diff
@@ -130,6 +134,67 @@ To apply the `jitter` over the entire dataset, use the 🤗 Datasets [`~datasets
...
+...     image = tf.image.random_contrast(image, 0.5, 2.0)
+...     image = tf.image.random_saturation(image, 0.75, 1.25)
+...     image = tf.image.random_hue(image, 0.1)
+...     image = tf.transpose(image, (2, 0, 1))
+...     return image
+
+
+>>> def transforms(image):
+...     image = tf.keras.utils.img_to_array(image)
+...     image = tf.transpose(image, (2, 0, 1))
```
Comment (amyeroberts) [inline docs/source/en/tasks/semantic_segmentation.mdx:165]: Just as a note (not a request for a change), you can also this in the image processor with:
`image_processor(images, labels, data_format="channels_first")`

**Review by amyeroberts — APPROVED** [review]
Nice - LGTM! Thanks for adding ❤️

File: docs/source/en/tasks/semantic_segmentation.mdx [hunk 2]
```diff
@@ -130,6 +134,67 @@ To apply the `jitter` over the entire dataset, use the 🤗 Datasets [`~datasets
 >>> test_ds.set_transform(val_transforms)
 ```
 
+</pt>
+</frameworkcontent>
+
+<frameworkcontent>
+<tf>
+It is common to apply some data augmentations to an image dataset to make a model more robust against overfitting.
+In this guide, you'll use `tf.image` to randomly change the color properties of an image, but you can also use any image
```
Comment (stevhliu) [inline docs/source/en/tasks/semantic_segmentation.mdx:143]: I'd add a link here so users can quickly reference it if it's the first time they're using `tf.image`.

```suggestion
In this guide, you'll use [`tf.image`](https://www.tensorflow.org/api_docs/python/tf/image) to randomly change the color properties of an image, but you can also use any image
```

File: docs/source/en/tasks/semantic_segmentation.mdx [hunk 3]
```diff
@@ -130,6 +134,67 @@ To apply the `jitter` over the entire dataset, use the 🤗 Datasets [`~datasets
...
+library you like.
+Define two separate transformation functions:
+- training data transformations that include image augmentation
+- validation data transformations that only transpose the images, since computer vision models in transformers expect channels-first layout
```
Comment (stevhliu) [inline docs/source/en/tasks/semantic_segmentation.mdx:147]:
```suggestion
- validation data transformations that only transpose the images, since computer vision models in 🤗 Transformers expect channels-first layout
```

**Review by stevhliu — APPROVED** [review]
Whoo, great job! Thanks for the addition 🔥

File: docs/source/en/tasks/semantic_segmentation.mdx (hunk 1 above)
Comment (MKhalusova) [reply docs/source/en/tasks/semantic_segmentation.mdx:165]: Thanks, @amyeroberts ! I didn't know I could do this.

File: docs/source/en/tasks/semantic_segmentation.mdx [hunk 4]
```diff
@@ -229,6 +333,112 @@ Once training is completed, share your model to the Hub with the [`~transformers
...
+...     columns=["pixel_values", "label"],
+...     shuffle=True,
+...     batch_size=batch_size,
+...     collate_fn=data_collator,
+... )
+```
+
+To compute the accuracy from the predictions and push your model to the 🤗 Hub, use [Keras callbacks](./main_classes/keras_callbacks).
+Pass your `compute_metrics` function to [KerasMetricCallback](./main_classes/keras_callbacks#transformers.KerasMetricCallback),
+and use the [PushToHubCallback](./main_classes/keras_callbacks#transformers.PushToHubCallback) to upload the model:
```
Comment (sgugger) [inline docs/source/en/tasks/semantic_segmentation.mdx:412]: Small nit, but linking to the doc with [\`KerasMetricCallback\`] and [\`PushToHubCallback\`] would make the links more resilient (in case we move those pages in the future).

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

File: docs/source/en/tasks/semantic_segmentation.mdx (hunk 4 above)
Comment (MKhalusova) [reply docs/source/en/tasks/semantic_segmentation.mdx]: Thank you, fixed :)
